### PR TITLE
Run3-alca201X Add the tree maker for IsoTrack analysis using the new calibration object

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Add the tree maker for IsoTrack analysis using the new calibration object

#### PR validation:

Tested using the cfg file in Calibration/HcalCalibAlgos/test/python

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Calibration/HcalCalibAlgos/plugins/HcalIsoTrackAnalyzer.cc [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+  const std::vector<unsigned int>*t_DetIds, *t_DetIds1, *t_DetIds3;
+  const std::vector<double>*t_HitEnergies, *t_HitEnergies1, *t_HitEnergies3;
+  const std::vector<bool>* t_trgbits;
+
+  unsigned int t_RunNo, t_EventNo;
+  bool t_TrigPass, t_TrigPassSel, t_L1Bit;
+  int t_Tracks, t_TracksProp, t_TracksSaved;
+  int t_TracksLoose, t_TracksTight, t_allvertex;
+  const std::vector<int>*t_ietaAll, *t_ietaGood, *t_trackType;
+  const std::vector<bool>* t_hltbits;
```
Comment (tvami) [inline Calibration/HcalCalibAlgos/plugins/HcalIsoTrackAnalyzer.cc:74]: @bsunanda I think these member variable should end with an _. Please see coding rule 2.9
"A trailing “_” is the preferred method to distinguish a data member from the getter method"

File: Calibration/HcalCalibAlgos/plugins/HcalIsoTrackAnalyzer.cc [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+  tree->Branch("t_qltyPVFlag", &t_qltyPVFlag, "t_qltyPVFlag/O");
+  tree->Branch("t_gentrackP", &t_gentrackP, "t_gentrackP/D");
+
+  t_DetIds = new std::vector<unsigned int>();
+  t_DetIds1 = new std::vector<unsigned int>();
+  t_DetIds3 = new std::vector<unsigned int>();
+  t_HitEnergies = new std::vector<double>();
+  t_HitEnergies1 = new std::vector<double>();
+  t_HitEnergies3 = new std::vector<double>();
+  t_trgbits = new std::vector<bool>();
```
Comment (tvami) [inline Calibration/HcalCalibAlgos/plugins/HcalIsoTrackAnalyzer.cc:254]: I think `new` is also discouraged, use `std::make_unique` instead or at least make sure that if you stay with the new there is a `delete` in the end to free the memory and avoid memory leaks. But my understanding is that CMS prefers to use the unique pointers instead

File: Calibration/HcalCalibAlgos/plugins/HcalIsoTrackAnalyzer.cc [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+  tree2->Branch("t_TracksLoose", &t_TracksLoose, "t_TracksLoose/I");
+  tree2->Branch("t_TracksTight", &t_TracksTight, "t_TracksTight/I");
+  tree2->Branch("t_TrigPass", &t_TrigPass, "t_TrigPass/O");
+  tree2->Branch("t_TrigPassSel", &t_TrigPassSel, "t_TrigPassSel/O");
+  tree2->Branch("t_L1Bit", &t_L1Bit, "t_L1Bit/O");
+  tree2->Branch("t_allvertex", &t_allvertex, "t_allvertex/I");
+  t_hltbits = new std::vector<bool>();
+  t_ietaAll = new std::vector<int>();
+  t_ietaGood = new std::vector<int>();
+  t_trackType = new std::vector<int>();
```
Comment (tvami) [inline Calibration/HcalCalibAlgos/plugins/HcalIsoTrackAnalyzer.cc:279]: Same thing about the `new` as above, see 7.7 of https://cms-sw.github.io/cms_coding_rules.html

File: Calibration/HcalCalibAlgos/plugins/HcalIsoTrackAnalyzer.cc (hunk 1 above)
Comment (bsunanda) [reply Calibration/HcalCalibAlgos/plugins/HcalIsoTrackAnalyzer.cc:74]: We have many macros which work with this naming convention (are used for past several years). Cannot change the names now

File: Calibration/HcalCalibAlgos/plugins/HcalIsoTrackAnalyzer.cc (hunk 2 above)
Comment (tvami) [reply Calibration/HcalCalibAlgos/plugins/HcalIsoTrackAnalyzer.cc:254]: ok, it that case, can you just make sure that everything is `delete`-ed in the end?

Comment (bsunanda) [reply Calibration/HcalCalibAlgos/plugins/HcalIsoTrackAnalyzer.cc]: I am removing the use of "new"

File: Calibration/HcalCalibAlgos/plugins/HcalIsoTrackAnalyzer.cc (hunk 1 above)
Comment (bsunanda) [reply Calibration/HcalCalibAlgos/plugins/HcalIsoTrackAnalyzer.cc]: Not changing that for backward compatibility of many macros and used by other tree analyzers

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by tvami** [comment]
please squash the last some commits and then I'll sign it

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by tvami** [comment]
+alca

**Pull request merged**